virtcontainers/qemu_arm64.go

```diff
@@ -38,15 +41,41 @@ var kernelRootParams = []Param{
 	{"root", "/dev/vda1"},
 }
 
+// variables rather than consts to allow tests to modify them
+var interruptFile = "/proc/interrupts"
+
+//on aarch64, we support different gic interrupt controllers
+//maximum number of vCPUs depends on the GIC version, or on how
+//many redistributors we can fit into the memory map.
+var gicList = map[string]uint32{
```

Comment:
I was reducing it from a few codes in `github.com/qemu/qemu/hw/arm/virt.c`. Related lines are line [L135](https://github.com/qemu/qemu/blob/stable-2.11/hw/arm/virt.c#L135) and line [L1306](https://github.com/qemu/qemu/blob/stable-2.11/hw/arm/virt.c#L1306). so maybe we can refer this file in comments.

Reply:
OK.. That is really dark magic. Please refer to these places so that we know where to look at if they need change in future.

And qemu doesn't seem to have gic version 4 ( [virt_get_gic_version ](https://github.com/qemu/qemu/blob/48a539df4a25b009c5a1239039349d54185fc0df/hw/arm/virt.c#L1632)). Is it added to be future proof?

Reply:
@bergwolf Yes, currently, the QEMU doesn't support GICv4, but we can use gic-version=3 for QEMU on host with GICv4. Just like we have done in runv for kvmtool. Because in some ways, GICv3 and GICv4 are the same. I also raised an issue about gic-version parameters of QEMU https://github.com/kata-containers/runtime/issues/614